lib/Optimizer/Dialect/Quake/QuakeOps.cpp

```diff
@@ -780,11 +781,62 @@ struct MergeRotationPattern : public OpRewritePattern<OP> {
     return success();
   }
 };
+
+// %4 = Something that returns float value
+// %5 = cc.alloca f64
+// cc.store %4, %5 : !cc.ptr<f64>
+// %6 = cc.load %5 : !cc.ptr<f64>
+// quake.ry (%6) %1 : (f64, !quake.ref) -> ()
+//
+// to
+//
+// quake.ry(%4) ...
```

Comment:
memtoreg should already do this.

Reply:
True. I can remove this if you think its ok to update the FileCheck tests because of the issue in my other comment.